src/App.js

```diff
@@ -1,16 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const toggleHeart = (id) => {
+    setChatData(
+      chatData.map((chatEntry) =>
+        chatEntry.id === id
+          ? { ...chatEntry, liked: !chatEntry.liked }
+          : chatEntry
+      )
+    );
+  };
```

Comment:
We can make the call to the `setChatData` to be less nested and easier to read by moving the logic around a bit. Because `map` returns a brand new list we can assign a variable to it and then pass it as an argument to `setChatdata`.

```javascript
  const toggleHeart = (id) => {
    let newChatData = chatData.map((chatEntry) =>
        chatEntry.id === id? { ...chatEntry, liked: !chatEntry.liked } : chatEntry
    );
      
    setChatData(newChatData);
  };
```